### ISSUE TITLE
Python-3.13.0b1 changes behavior on exec() versus Python-3.12 and before ... is it normal ?

### ISSUE THREAD

**Issue by stonebig** [python/cpython#118938]
# Bug report

### Bug description:

this code works on Python-3.12 and gives
````
Hello, a World !
Hello, b World !
````

on Python-3.13.0b1 it fails on the b part:
````pytb
Hello, a World !
Traceback (most recent call last):
  File "C:/..../python-3.13.0b1.amd64/Lib/site-packages/sqlite_bro/testb.py", line 16, in <module>
    test()
  File "C:/.../python-3.13.0b1.amd64/Lib/site-packages/sqlite_bro/testb.py", line 14, in test
    be=eval('py_hellob');
  File "<string>", line 1, in <module>
    __import__('idlelib.run').run.main(True)
NameError: name 'py_hellob' is not defined. Did you mean: 'py_helloa'?
````

```python
a="""def py_helloa():
    "hello world"
    return ("Hello, a World !");""";

exec(a, globals(), locals());
ae=eval('py_helloa');
print(ae())

def test():
    b="""def py_hellob():
    "hello world"
    return ("Hello, b World !");""";
    exec(b, globals(), locals());
    be=eval('py_hellob');
    print(be())
test()

```

### CPython versions tested on:

3.12, 3.13

### Operating systems tested on:

Windows

**Comment by hugovk** [comment]
Thanks for testing the beta! 

Perhaps related to https://github.com/python/cpython/issues/118888?

**Comment by terryjreedy** [comment]
I am rather sure this is a duplicate.
@gaogaotiantian

**Comment by stonebig** [comment]
I had read PEP 667, looks logically nice, but how to rewrite my code ?

**Comment by stonebig** [comment]
seems to work in Python-3.12 too...

**Comment by stonebig** [comment]
so I did a fix to work on Python-3.13.0b1 for my package, but it's not clear if I should have done it now, with https://github.com/python/cpython/issues/118921

**Comment by terryjreedy** [comment]
Further discussion should be on the original issue #118888.
